extensions/wikia/WikiFactory/Dumps/DumpsOnDemand.php

```diff
@@ -68,15 +63,19 @@ static public function customSpecialStatistics( &$specialpage, &$text ) {
 
 		$bIsAllowed = $wgUser->isAllowed( 'dumpsondemand' ) && !in_array( $wgUser->getName(), $aDumpRequestBlacklist );
 		$tmpl->set( 'bIsAllowed', $bIsAllowed );
+		$tmpl->set( 'editToken', $wgUser->getEditToken());
 
-		$text .= $tmpl->render( "dod" );
-
-		if( $wgRequest->wasPosted() && $bIsAllowed ) {
+		if ( $wgRequest->wasPosted() && $available && $bIsAllowed && $wgUser->matchEditToken( $wgRequest->getVal( 'editToken' ) ) ) {
 			self::queueDump( $wgCityId );
 			wfDebug( __METHOD__, ": request for database dump was posted\n" );
 			$text = Wikia::successbox( wfMsg( "dump-database-request-requested" ) ) . $text;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)